Src/BootCamp.Chapter/BalanceStats.cs

```diff
@@ -7,31 +10,232 @@ public static class BalanceStats
         /// </summary>
         public static string FindHighestBalanceEver(string[] peopleAndBalances)
         {
-            return "";
+            if (peopleAndBalances == null || peopleAndBalances.Length == 0)
+            {
+                return "N/A.";
+            }
+
+            var HighestBalanceEver = decimal.MinValue;
+            var PersonWithHighestBalance = new StringBuilder();
+
+            for (int i = 0; i <= peopleAndBalances.Length - 1; i++)
+            {
+                var currentPersonData = peopleAndBalances[i].Split(',');
+                var highestAmountOfPerson = decimal.Parse(currentPersonData[1..].Max());
+
+                if (highestAmountOfPerson > HighestBalanceEver)
```

Comment:
Could use a function. SetNewMaximum

Reply:
oke, so instead of the Max of Linq , use a function  setNewMaximum to find the max amount of a person. Do I understand you well ?

Reply:
No, it's not just that. If you need currentPersonData just for finding the highestamoutOfPerson, you might as well keep both in one function. Also please do first and ask later.

Reply:
Oke,  I will make what I think you are trying to explain to me and make a new PR.

Reply:
sorry. but I thik we still misunderstood each other. What is the purpose of the SetNewMaximum method where you talk about.  I can make a function that replaces the linq part but what should this function now exactly do.